addons/udes_stock_refactoring/models/stock_move.py

```diff
@@ -359,6 +359,132 @@ def _refactor_action_by_maximum_quantity(self, maximum_qty):
 
         return self | new_pickings.move_lines
 
+    def _refactor_action_by_maximum_weight(self, maximum_weight):
+        """
+        Split move_lines out into pickings with a maximum weight
+        This first tries to create pickings with a single move with the maximum allowed
+        Then combines the remaining moves into pickings with maximum allowed
+        """
+        StockMove = self.env["stock.move"]
+
+        if maximum_weight < 1:
+            raise ValidationError(_("Cannot split quants into weight: %i") % maximum_weight)
+
+        all_moves = StockMove.browse()
+
+        for picking, moves in self.groupby("picking_id"):
+            moves_to_refactor = []
+            grouped_moves = moves._split_and_group_by_weight(maximum_weight)
+            max_range = len(grouped_moves)
+            # Build iterable of grouped moves to refactor, which can be passed into refactor_by_move_groups.
+            # Give each set of moves a unique group name so the pickings are kept separate.
+            for idx, moves_to_keep in enumerate(grouped_moves[:max_range]):
+                group_name = f"{picking.name}-{idx:0>3}"
+                moves_to_refactor.insert(0, (group_name, moves_to_keep))
+                all_moves |= moves_to_keep
+
+            all_moves |= moves.refactor_by_move_groups(moves_to_refactor)
+        return all_moves
+
+    def _split_and_group_by_weight(self, maximum_weight):
+        """
+        Split moves into groups of up to a maximum weight
+        :param maximum_weight: float weight to split and group moves
+        :returns: list of grouped moves
+        """
+        grouped_moves = []
+
+        remaining_moves = self
+
+        # See if any mls are equal to the maximum and add them as individual groups
+        exact_moves = self.filtered(
```

Comment:
Question: Do we expect to have moves exactly with maximum_height frequently, so that you are trying to handle in a separate filter?

Reply:
it's just copying how refactor by weight works tbh, and it seemed like a sane check to remove exact matches first before continuing with partial matches. I could probably change this to >= maximum_weight and handle over weight items (as they need to be in their own pick) here trather than inside `_moves_for_weight`